### PR TITLE
Move CLI from server/pxf-cli to cli

### PR DESCRIPTION
**Pull request by lavjain**
Co-authored-by: Oliver Albertini <oalbertini@pivotal.io>
Co-authored-by: Lav Jain <ljain@pivotal.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+PXF_VERSION := $(shell grep '^version=' ./server/gradle.properties | cut -d "=" -f2)
```
Comment (denalex) [inline Makefile]: wonder if we can do it vice-versa -- have the version in the version file at the top level and have GRadel read it from there. Perhaps for the next iteration ?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+ifneq "$(PXF_HOME)" ""
+	BUILD_PARAMS+= -DdeployPath="$(PXF_HOME)"
+else ifneq "$(GPHOME)" ""
+	PXF_HOME= "$(GPHOME)/pxf"
+	BUILD_PARAMS+= -DdeployPath="$(PXF_HOME)"
```
Comment (denalex) [inline Makefile]: this line can be moved out of the statement and have if statement only set PXF_HOME if it is not yet set

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+server:
+	make -C server
+
+clean:
+	make -C cli/go/src/pxf-cli clean
+	make -C server clean
+
+tar:
+	make -C cli/go/src/pxf-cli tar
+	make -C server tar
```
Comment (denalex) [inline Makefile]: so this will produce 2 different tarballs ? How and when will they be combined to produce a releasable artifact ?

Comment (oliverralbertini) [reply Makefile]: This is a two-step process that will make one tarball. The cli steps makes the go artifact and renames the shell script (pxf-service). Then the server tar step uses gradle to make the releasable artifact.

**Pull request merged**